WcaOnRails/lib/results_validators/competitor_limit_validator.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ResultsValidators
+  class CompetitorLimitValidator < GenericValidator
+    COMPETITOR_LIMIT_WARNING = "The number of persons in the competition (%{n_competitors}) is above the competitor limit (%{competitor_limit})."\
+      " Unless a specific agreement was made when announcing the competition (such as a per-day competitor limit), the results of the competitors registered after the competitor limit was reached must be removed."
+
+    @@desc = "For competition with a competitor limit, this validator checks that this limit is respected."
+
+    def validate(competition_ids: [], model: Result, results: nil)
+      reset_state
+
+      if competition_ids.empty?
+        competition_ids = results.map(&:competitionId).uniq
+      end
```

Comment:
Ah actually, why do we set competition ids from results here, I believe in most places it's the other way round: https://github.com/thewca/worldcubeassociation.org/blob/206794e0935e18b9d7373fca8b7279bb837e7d5f/WcaOnRails/lib/results_validators/scrambles_validator.rb#L14-L15

Reply:
We don't actually need any result for this check, we just want a list of competitions!
Either `competition_ids` are provided and we check these competitions, or some `results` are provided from which we want to extract the corresponding competitions.

Reply:
Ah, because we query results anyway, gotcha!